### ISSUE TITLE
[Firestore] New project doesn't respond when setting data

### ISSUE THREAD

**Issue by V1rus999** [FirebaseExtended/flutterfire#2827]
Hi.

I created a brand new  flutter application with a backing (new) firebase account. Up to now I have gotten firebase auth to work fine. I am now looking at adding firestore. 

I haven't added a firestore DB at all, but at the moment I'm just writing some code to push a user to the firestore DB and I am expecting it to fail.

Problem: The Future<void> from setData never returns. 

Here is my code:

```
class Database {
  final CollectionReference _usersCollectionReference =
      Firestore.instance.collection('users');

  Future<void> addUser(User user) async {
    try {
      print("Starting addUser request");
      print(user.toMap());
      final res = await _usersCollectionReference
          .document(user.uid)
          .setData(user.toMap());
      print("Got response");
      return res;
    } catch (e) {
      print("addUser error");
      print(e);
    }
  }
}
```

I never see the "Got response" or "addUser error" logs. At the very least I expect an exception of some sort. 

Is this expected behaviour? What am I missing?

Flutter version 1.20.0-0.0.pre
cloud_firestore: 0.13.6
Phones: Huawei Y7 and an emulator
I assume the android side has been set up correctly seeing that Firebase auth works.

**Comment by V1rus999** [comment]
I haven't done any setup, but again, I am expecting this to fail in some way.

**Comment by V1rus999** [comment]
**First time I run  this function I get the following logs:**

```
I/flutter (13978): Starting addUser request
I/flutter (13978): {displayName: test, email: test@mail.com, uid: test123}
W/DynamiteModule(13978): Local module descriptor class for providerinstaller not found.
I/DynamiteModule(13978): Considering local module providerinstaller:0 and remote module providerinstaller:0
W/ProviderInstaller(13978): Failed to load providerinstaller module: No acceptable module found. Local version is 0 and remote version is 0.
D/ApplicationLoaders(13978): Returning zygote-cached class loader: /system/framework/android.hidl.base-V1.0-java.jar
D/ApplicationLoaders(13978): Returning zygote-cached class loader: /system/framework/android.hidl.manager-V1.0-java.jar
D/ApplicationLoaders(13978): Returning zygote-cached class loader: /system/framework/android.hidl.base-V1.0-java.jar
W/mple.google_ss(13978): Accessing hidden method Lsun/misc/Unsafe;->getLong(Ljava/lang/Object;J)J (greylist,core-platform-api, linking, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/misc/Unsafe;->arrayBaseOffset(Ljava/lang/Class;)I (greylist,core-platform-api, linking, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/misc/Unsafe;->copyMemory(JJJ)V (greylist, linking, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/misc/Unsafe;->objectFieldOffset(Ljava/lang/reflect/Field;)J (greylist,core-platform-api, linking, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/misc/Unsafe;->getByte(J)B (greylist,core-platform-api, linking, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/misc/Unsafe;->getByte(Ljava/lang/Object;J)B (greylist,core-platform-api, linking, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/misc/Unsafe;->getLong(J)J (greylist,core-platform-api, linking, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/misc/Unsafe;->putByte(JB)V (greylist,core-platform-api, linking, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/misc/Unsafe;->putByte(Ljava/lang/Object;JB)V (greylist,core-platform-api, linking, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/misc/Unsafe;->getLong(Ljava/lang/Object;J)J (greylist,core-platform-api, reflection, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/misc/Unsafe;->getLong(Ljava/lang/Object;J)J (greylist,core-platform-api, reflection, allowed)
W/mple.google_ss(13978): Accessing hidden field Ljava/nio/Buffer;->address:J (greylist, reflection, allowed)
I/mple.google_ss(13978): The ClassLoaderContext is a special shared library.
I/chatty  (13978): uid=10218(com.example.google_sso) AsyncTask #1 identical 2 lines
I/mple.google_ss(13978): The ClassLoaderContext is a special shared library.
W/mple.google_ss(13978): Accessing hidden field Ldalvik/system/BaseDexClassLoader;->pathList:Ldalvik/system/DexPathList; (greylist, reflection, allowed)
W/mple.google_ss(13978): Accessing hidden field Ldalvik/system/DexPathList;->nativeLibraryDirectories:Ljava/util/List; (greylist, reflection, allowed)
W/mple.google_ss(13978): Accessing hidden field Ldalvik/system/DexPathList;->systemNativeLibraryDirectories:Ljava/util/List; (greylist, reflection, allowed)
W/mple.google_ss(13978): Accessing hidden field Ldalvik/system/DexPathList;->nativeLibraryPathElements:[Ldalvik/system/DexPathList$NativeLibraryElement; (greylist, reflection, allowed)
W/mple.google_ss(13978): Accessing hidden method Ldalvik/system/DexPathList;->makePathElements(Ljava/util/List;)[Ldalvik/system/DexPathList$NativeLibraryElement; (greylist, reflection, allowed)
V/NativeCrypto(13978): Registering com/google/android/gms/org/conscrypt/NativeCrypto's 284 native methods...
W/mple.google_ss(13978): Accessing hidden method Ljava/security/spec/ECParameterSpec;->getCurveName()Ljava/lang/String; (greylist, reflection, allowed)
D/NetworkSecurityConfig(13978): No Network Security Config specified, using platform default
I/ProviderInstaller(13978): Installed default security provider GmsCore_OpenSSL
W/mple.google_ss(13978): Accessing hidden field Ljava/net/Socket;->impl:Ljava/net/SocketImpl; (greylist, reflection, allowed)
W/mple.google_ss(13978): Accessing hidden method Ldalvik/system/CloseGuard;->get()Ldalvik/system/CloseGuard; (greylist,core-platform-api, linking, allowed)
W/mple.google_ss(13978): Accessing hidden method Ldalvik/system/CloseGuard;->open(Ljava/lang/String;)V (greylist,core-platform-api, linking, allowed)
W/mple.google_ss(13978): Accessing hidden field Ljava/io/FileDescriptor;->descriptor:I (greylist, JNI, allowed)
W/mple.google_ss(13978): Accessing hidden method Ljava/security/spec/ECParameterSpec;->setCurveName(Ljava/lang/String;)V (greylist, reflection, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/security/x509/AlgorithmId;->get(Ljava/lang/String;)Lsun/security/x509/AlgorithmId; (greylist, reflection, allowed)
W/mple.google_ss(13978): Accessing hidden method Lsun/security/x509/AlgorithmId;->getName()Ljava/lang/String; (greylist,core-platform-api, reflection, allowed)
W/mple.google_ss(13978): Accessing hidden method Ldalvik/system/BlockGuard;->getThreadPolicy()Ldalvik/system/BlockGuard$Policy; (greylist,core-platform-api, linking, allowed)
W/mple.google_ss(13978): Accessing hidden method Ldalvik/system/BlockGuard$Policy;->onNetwork()V (greylist, linking, allowed)
```
**On the second run of the function I only see this and no more output**

```
I/flutter (13978): Starting addUser request
I/flutter (13978): {displayName: test, email: test@mail.com, uid: test123}
```

**Comment by TahaTesser** [comment]
Hi @V1rus999 

Just tried on latest `dev` channel using `cloud_firestore: ^0.13.7`, no issues on first run 

<details> 
<summary>Code Sample</summary> 

``` 
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Firestore Example'),
      ),
      floatingActionButton: FloatingActionButton.extended(
          label: Text('Add User'),
          icon: Icon(Icons.add),
          onPressed: () {
            Database().addUser();
          }),
    );
  }
}

class Database {
  final CollectionReference _usersCollectionReference =
      Firestore.instance.collection('users');

  Future<void> addUser() async {
    try {
      print("Starting addUser request");
      // print(user.toMap());
      final res = await _usersCollectionReference
          .document('user')
          .setData({'id': '124'});
      print("Got response");
      return res;
    } catch (e) {
      print("addUser error");
      print(e);
    }
  }
}

``` 

</details> 


<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, 1.20.0-1.0.pre, on Mac OS X 10.15.5 19F101, locale
    en-GB)
    • Flutter version 1.20.0-1.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision f73f498da1 (8 days ago), 2020-06-18 08:23:22 -0700
    • Engine revision c0d52b98d1
    • Dart version 2.9.0 (build 2.9.0-16.0.dev 2b917f5b6a)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.46.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.11.0

[✓] Connected device (6 available)
    • SM M305F      • 32003c30dc19668f                     • android-arm64  •
      Android 10 (API 29)
    • Taha’s iPhone • 00008020-001059882212002E            • ios            •
      iOS 13.5.1
    • iPhone 11     • 0EC80516-8F99-4A0C-89ED-2273579862F9 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)
    • macOS         • macOS                                • darwin-x64     •
      Mac OS X 10.15.5 19F101
    • Web Server    • web-server                           • web-javascript •
      Flutter Tools
    • Chrome        • chrome                               • web-javascript •
      Google Chrome 83.0.4103.116

• No issues found!

``` 

</details>

**Comment by Salakar** [comment]
Hey, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
